### DISCUSSION TITLE
Users list / groups management

### DISCUSSION THREAD

**Discussion by AlexandreBeauduin** [slskd/slskd#709] [Ideas]
Hello,

First of all, thanks for slskd :) 
Just discovered it and I like it :) it'll replace a win XP VM :)

I was wondering if it woud be possible to add a user list/group management in the webui instead of having to do it in the yaml configuration.
And also adding user to a list or group from the search, download and upload pages ?

Thanks :)

pudeldoodle

**Comment by jpdillingham** [comment]
Thanks for the suggestion!

The configuration is done the way it is currently for a few reasons:

* Maintaining the sort of UI that would be required would slow feature development down considerably
* The run-time configuration is primarily handled via the YAML file, and reading and writing the YAML file programmatically (as would be needed to change things through the UI) would cause the file to be re-formatted, and comments would be lost when this happens

For the foreseeable future I don't see myself putting any effort towards this, but I'd be thrilled if someone else wants to take up the work.

If I manage to find a YAML library for .NET that can programmatically edit files without destroying comments then I'll reconsider.  I think the easier route forward would be to provide a new layer of configuration that would override the YAML file, and use that exclusively for run-time changes.
